site/src/pages/HealthPage/HealthPage.tsx

```diff
@@ -138,26 +79,101 @@ export function HealthPageView({
           overflow: "hidden",
         }}
       >
-        <Box
-          sx={{
+        <div
+          css={{
             width: 256,
             flexShrink: 0,
-            borderRight: (theme) => `1px solid ${theme.palette.divider}`,
+            borderRight: `1px solid ${theme.palette.divider}`,
+            fontSize: 14,
           }}
         >
-          <Box
-            sx={{
-              fontSize: 10,
-              textTransform: "uppercase",
-              fontWeight: 500,
-              color: (theme) => theme.palette.text.secondary,
-              padding: "12px 24px",
-              letterSpacing: "0.5px",
+          <div
+            css={{
+              padding: 24,
+              display: "flex",
+              flexDirection: "column",
+              gap: 16,
             }}
           >
-            Health
-          </Box>
-          <Box component="nav">
+            <div>
+              <div
+                css={{
+                  display: "flex",
+                  alignItems: "center",
+                  justifyContent: "space-between",
+                }}
+              >
+                <CheckCircleOutlined
+                  css={{
+                    width: 32,
+                    height: 32,
+                    color: theme.palette.success.light,
+                  }}
+                />
+                <Tooltip title="Refresh health checks">
+                  <IconButton
+                    size="small"
+                    disabled={isRefreshing}
+                    data-testid="healthcheck-refresh-button"
+                    onClick={forceRefresh}
+                  >
+                    {isRefreshing ? (
+                      <CircularProgress size={16} />
+                    ) : (
+                      <ReplayIcon css={{ width: 20, height: 20 }} />
+                    )}
+                  </IconButton>
+                </Tooltip>
+              </div>
+              <div css={{ fontWeight: 500, marginTop: 16 }}>
+                {healthStatus.healthy ? "Healthy" : "Unhealthy"}
+              </div>
+              <div
+                css={{
+                  color: theme.palette.text.secondary,
+                  lineHeight: "150%",
+                }}
+              >
+                {healthStatus.healthy
+                  ? Object.keys(sections).some(
+                      (key) =>
+                        healthStatus[key as keyof typeof sections].warnings !==
+                          null &&
```

Comment:
If the key doesn't exist on `healthStatus`, wouldn't `healthStatus[key].warnings` become `undefined` instead of `null`?

Reply:
the backend returns null when there are no warnings.